crates/runtime/src/env/mod.rs

```diff
@@ -54,13 +52,16 @@ pub trait EnvTypes {
     type AppStore: AppStore;
 
     /// Compute `Template` address type.
-    type TemplateAddressCompute: TemplateAddressCompute;
+    type TemplateAddressCompute: AddressLocator<Template, Address = TemplateAddr>;
 
     /// Compute `App` address type.
-    type AppAddressCompute: AppAddressCompute;
+    type AppAddressCompute: AddressLocator<ExtSpawnApp, Address = AppAddr>;
 
     /// `Template` content Hasher type.
     type TemplateHasher: TemplateHasher;
+
+    /// A pricing engine for templates.
+    type Pricer: PriceResolver;
```

Comment:
See my remark above about inject a resolver.

It's unlikely that we'll want to change the way we compute `TemplateHash` or an `Address` of a `Template` or `Account`. But it's almost certain we'll have versioning in regard to `PriceReslover`.

Reply:
Partially addressed by the `V0` rename. Resolving in favor of the conversion above.